### PR TITLE
Add weapon_rope

### PR DESCRIPTION
**Pull request by lewa-j**

### CODE REVIEW

File: dlls/weapons.cpp [hunk 1]
```diff
@@ -363,6 +363,7 @@ void W_Precache( void )
 		UTIL_PrecacheOtherWeapon( "weapon_big_cock" );
 	if( cvar_allow_gateofbabylon.value )
 		UTIL_PrecacheOtherWeapon( "weapon_gateofbabylon" );
+	UTIL_PrecacheOtherWeapon( "weapon_rope" );
```
Comment (a1batross) [inline dlls/weapons.cpp]: Maybe put under cvar too?

File: dlls/Makefile.mingw [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+BASE_CFLAGS= -DCLIENT_WEAPONS -Dstricmp=strcasecmp -Dstrnicmp=strncasecmp -D_snprintf=snprintf \
+	-fno-exceptions -DNO_VOICEGAMEMGR -Wno-conversion-null -Wno-write-strings -std=gnu++98 -Wno-invalid-offsetof
+
+
+#safe optimization
+CFLAGS=$(BASE_CFLAGS) -m32 -w -O1
+
+#full optimization
+#CFLAGS=$(BASE_CFLAGS) -w -O1 -m486 -ffast-math -funroll-loops \
```
Comment (a1batross) [inline dlls/Makefile.mingw]: >full optimization

>-O1 -m486

LOL.

File: dlls/swinging_rope.cpp [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+#include "util.h"
+#include "cbase.h"
+#include "player.h"
+#include "customentity.h"
+#include "customweapons.h"
+#include "unpredictedweapon.h"
+
+class CPiton;
+
+class RopeGun : public CBasePlayerWeaponU
```
Comment (a1batross) [inline dlls/swinging_rope.cpp]: CRopeGun.

File: dlls/Makefile.mingw (hunk 2 above)
Comment (nekonomicon) [reply dlls/Makefile.mingw]: -ffast-math is better =_=

Comment (lewa-j) [reply dlls/Makefile.mingw]: Just copypasted from dlls/Makefile

Comment (nekonomicon) [reply dlls/Makefile.mingw]: Ah... Ok.

Comment (nekonomicon) [reply dlls/Makefile.mingw]: Unified Makefile would be better anyway.

File: dlls/weapons.cpp (hunk 1 above)
Comment (lewa-j) [reply dlls/weapons.cpp]: Done

File: dlls/swinging_rope.cpp (hunk 3 above)
Comment (lewa-j) [reply dlls/swinging_rope.cpp]: Done too

File: engine/eiface.h [hunk 4]
```diff
@@ -359,7 +359,7 @@ typedef enum _fieldtypes
 #define offsetof(s,m)	(size_t)&(((s *)0)->m)
 #endif
 
-#define _FIELD(type,name,fieldtype,count,flags)		{ fieldtype, #name, offsetof(type, name), count, flags }
+#define _FIELD(type,name,fieldtype,count,flags)		{ fieldtype, #name, (int) offsetof(type, name), count, flags }
```
Comment (nekonomicon) [inline engine/eiface.h]: It's not a good idea.
And it was fixed in master branch.
https://github.com/FWGS/hlsdk-xash3d/commit/cf6647e07c452f6522aa8b5a9619a1ac1246f9af

Comment (lewa-j) [reply engine/eiface.h]: OK

**Pull request closed without merging**